lib/host-id.c

```diff
@@ -51,13 +70,18 @@ host_id_init(PersistState *state)
   PersistEntryHandle handle;
   HostIdState *host_id_state;
   gboolean new_host_id_required = FALSE;
+  gboolean use_legacy_host_id = FALSE;
```

Comment:
For me it is a little hard to follow because the new code is interleaving with the original code in a complex way. There are also a lot of states here in the machinery: new_host_id_required, use_legacy_host_id, legacy_host_id, global_host_id.

I am thinking of reorganizing this function as follows:

```
gboolean
host_id_init(PersistState *state)
{
  convert_legacy_host_id_entry(...);
  [... leave everything as it is now].
}
```

Whereas `convert_legacy_host_entry`:
- lookup the legacy key. If not available, just return.
- lookup the new key. If available, just delete the old entry and return: the damage is already done, we do not want to change host_id that is currently used.
- Rename the entry.

From that we can do everything as usual: we are either having a proper host id entry or need to create a fresh one from scratch. Hence the rest of host_id_init can stay.

What do you think?

Reply:
I feel a dissonance in a clean code perspective with:
```
  if (new_host_id_required)
    {
      if (_load_host_id_from_legacy_persist_entry(state, &legacy_hostid))
```
If a new `host_id` is required, why are we loading it? I understand the algorithm here, just the names/order are confusing.

furiel's refactor idea seems nice to me.

Reply:
Personally, I am okay with the new version.

Reply:
We discussed with @szemere and concluded that we would like to promote the happy path: no conversion is needed. In my version, the conversion would be checked each time. In summary, I'm fine with the new version.